src/PhpSpec/Locator/PSR0/PSR0Locator.php

```diff
@@ -357,7 +357,7 @@ private function createResourceFromSpecFile($path)
      */
     private function validatePsr0Classname($classname)
     {
-        $pattern = '/^([a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*[\/\\\\]?)*[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*$/';
+        $pattern = '/^([a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*[\/\\\\]?)*[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*\z$/';
```

Comment:
You don't need the dollar at the end if you've added `\z`. You can replace the `^` at the start with `\A` too